embassy-rp/src/spi.rs

```diff
@@ -244,6 +244,27 @@ impl<'d, T: Instance, M: Mode> Spi<'d, T, M> {
         // enable
         p.cr1().write(|w| w.set_sse(true));
     }
+
+    /// Set SPI config.
+    pub fn set_config(&mut self, config: Config) {
+        let (presc, postdiv) = calc_prescs(config.frequency);
+        
+        let p = self.inner.regs();
+        // disable
+        p.cr1().write(|w| w.set_sse(false));
+
+        // change stuff
+        p.cpsr().write(|w| w.set_cpsdvsr(presc));
+        p.cr0().write(|w| {
```

Comment:
this is copypasted from `new_inner`. could you extract it to a private function `fn apply_config()` instead?

Reply:
After the CI failure, I see that the `SetConfig` trait in the [embassy-embedded-hal](https://docs.embassy.dev/embassy-embedded-hal/git/default/trait.SetConfig.html) actually does what I want, so this change may not be needed. Should we consider changing the trait implementation to disable/reenable the peripheral as the `set_frequency` method does?

Reply:
whatever's needed for it to work. I don't know if it's required

Reply:
The [Pico SDK disables the driver before changing frequencies](https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_spi/spi.c), as does the `set_frequency` method. I've updated to add a private `apply_config` method and then reused the `Spi` struct's `set_config` method in the trait implementation.

Reply:
This would also be a breaking change as the method's could not conflict depending on Trait in scope or not - I propose we just remove the proposed Spi `set_config` method and update the Trait implementation to disable/reenable the driver at the expense of some code duplication - I'll wait for your thoughts before touching again. Thanks for all your work.

Reply:
don't worry about breaking changes, nextreleasewill be breaking anyway due to other changes.